### PR TITLE
Fix/remove warning in test

### PR DESCRIPTION
**Pull request by samouss**
**Summary**

This PR aims to fixed all the warnings from the tests & PropTypes (when found).

**Result**

You can check the ouput of the on Travis.

**Before**:

screen shot 2017-10-12 at 13 02 21

**After**:

screen shot 2017-10-12 at 12 58 27

### CODE REVIEW

File: src/components/Slider/Slider.js [hunk 1]
```diff
@@ -48,9 +48,7 @@ export class RawSlider extends Component {
       min,
       ...times(steps - 1, step => min + stepsLength * (step + 1)),
       max,
-    ]
-      // bug with `key={ 0 }` and preact, see https://github.com/developit/preact/issues/642
-      .map(pitPoint => (pitPoint === 0 ? 0.000001 : pitPoint));
```
Comment (samouss) [inline src/components/Slider/Slider.js]: It's already fixed in Rheostat fork (see: [rheostat/Slider.jsx#L665](https://github.com/algolia/rheostat/blob/dist-preact/src/Slider.jsx#L665)).

File: package.json [hunk 2]
```diff
@@ -74,7 +74,7 @@
     "eslint-plugin-prettier": "2.3.1",
     "eslint-plugin-react": "7.3.0",
     "expect": "1.20.2",
-    "expect-jsx": "2.6.0",
+    "expect-jsx": "4.0.0",
```
Comment (bobylito) [inline package.json]: That's already one thing that we won't have to do using renovate :)

File: src/components/Slider/Pit.js [hunk 3]
```diff
@@ -30,8 +31,12 @@ const Pit = ({ key, style, children }) => {
 };
 
 Pit.propTypes = {
-  key: PropTypes.string,
```
Comment (bobylito) [inline src/components/Slider/Pit.js]: Why isn't the key needed anymore?

File: package.json (hunk 2 above)
Comment (samouss) [reply package.json]: Yep, but the warnings came from this package. So how we handle that? We use renovate or we can upgrade the dependency manually?

File: src/components/Slider/Pit.js (hunk 3 above)
Comment (samouss) [reply src/components/Slider/Pit.js]: Actually we used the `key` to bypass a bug with `preact`. Now it seems to render correctly, I  don't spot any bugs. I know @iam4x worked on it, if you think that something goes wrong don't hesitate :)

Comment (iam4x) [reply src/components/Slider/Pit.js]: It should be good, wasn't too sure about this fix I've made. It removed a warning and it was also used when I tried to moved the tests to a full preact rendering.

File: src/widgets/numeric-refinement-list/__tests__/numeric-refinement-list-test.js [hunk 4]
```diff
@@ -130,7 +130,7 @@ describe('numericRefinementList()', () => {
           header: '',
           // eslint-disable-next-line max-len
           item:
-            '<label class="{{cssClasses.label}}">\n <input type="radio" class="{{cssClasses.radio}}" name="{{attributeName}}" {{#isRefined}}checked{{/isRefined}} />{{label}}\n</label>',
+            '<label class="{{cssClasses.label}}">\n  <input type="radio" class="{{cssClasses.radio}}" name="{{attributeName}}" {{#isRefined}}checked{{/isRefined}} />{{label}}\n</label>',
```
Comment (iam4x) [inline src/widgets/numeric-refinement-list/__tests__/numeric-refinement-list-test.js]: Hmmm, this is not really expected isn't it?

Comment (samouss) [reply src/widgets/numeric-refinement-list/__tests__/numeric-refinement-list-test.js]: Yes it is, in fact due it's to the indentation in `defaultTemplate` ([link](https://github.com/algolia/instantsearch.js/blob/develop/src/widgets/numeric-refinement-list/defaultTemplates.js#L5)). You can see that at the begining of the line we have two spaces.

Comment (iam4x) [reply src/widgets/numeric-refinement-list/__tests__/numeric-refinement-list-test.js]: But, this is not in the PR?

Comment (samouss) [reply src/widgets/numeric-refinement-list/__tests__/numeric-refinement-list-test.js]: I just need to change the test, it's probably due some internal change in `react-element-to-jsx-string` that is used by `expect-jsx`.

File: package.json (hunk 2 above)
Comment (bobylito) [reply package.json]: That's actually a good thing to not have to do it with Renovate. That's probably a strategy that we should adopt more often 👍

**Pull request merged**